lib/context_request_middleware/context/cookie_session_retriever.rb

```diff
@@ -53,8 +55,7 @@ def session_id
       end
 
       def req_cookie_session_id
-        Rack::Utils.parse_cookies(@request.env)['_session_id'] ||
-          (@request.env['action_dispatch.cookies'] || {})['_session_id']
+        session
```

Comment:
This also sounds odd from naming. You are returning the session whereas the name suggest differently.